ext/session/session.c

```diff
@@ -485,19 +485,24 @@ PHPAPI int php_session_valid_key(const char *key) /* {{{ */
 /* }}} */
 
 
-static void php_session_gc(void) /* {{{ */
+static zend_long php_session_gc(zend_bool immediate) /* {{{ */
 {
```

Comment:
I updated patch so that it can simply use zend_long. 

User defined handlers are able to return long value and FALSE. It may return  BOOL or LONG which is set to number of deleted sessions.